### PR TITLE
feat: add display of 3d mesh

### PR DESCRIPTION
**Pull request by anna-charlotte**
Goals:

We want to add 3d mesh and point cloud to display(), i.e. if you have a Document instance on which you have called .load_uri_to_point_cloud_tensor() or load_uri_to_vertices_and_faces, we want to display an interactive 3d plot, when calling display() on the instance. Similar to the one shown here in cell 10: https://trimsh.org/examples.quick_start.html

- [x] add `.display()` for 3d mesh
- [x] check and update documentation, if required. See [guide](https://github.com/jina-ai/jina/blob/master/CONTRIBUTING.md#documentation-guidelines)

### CODE REVIEW

**Review by NicholasDunham — APPROVED** [review]

**Comment by samsja** [comment]
> code looks great for now. Lets add the display function to validate that the representation is good
and lets do it in this PR so lets not merge before adding it

File: docarray/document/mixins/plot.py [hunk 1]
```diff
@@ -92,7 +96,65 @@ def display(self, from_: Optional[str] = None):
...
+        Tells if Document stores a 3D object saved as point cloud or vertices and face.
+        :return: bool.
+        """
+        if self.uri and self.uri.endswith(tuple(MESH_FILE_EXTENSIONS)):
+            return True
+        elif self.tensor is not None and self.tensor.shape[1] == 3:
+            return True
+        elif self.chunks is not None:
+            name_tags = [c.tags['name'] for c in self.chunks]
+            if 'vertices' in name_tags and 'faces' in name_tags:
```
Comment (JoanFM) [inline docarray/document/mixins/plot.py:110]: let's have these `str` as CONSTANTS somewhere in the mesh mixin

File: docarray/document/mixins/plot.py [hunk 2]
```diff
@@ -92,7 +96,65 @@ def display(self, from_: Optional[str] = None):
...
+    def display_3d(self) -> None:
+        """Plot 3d data."""
+        from IPython.display import display
+        import trimesh
+
+        if self.tensor is not None:
+            # point cloud from tensor
+            from IPython import get_ipython
+
+            in_notebook = get_ipython().__class__.__name__ == 'ZMQInteractiveShell'
```
Comment (JoanFM) [inline docarray/document/mixins/plot.py:124]: also a CONSTANT here, I know is something we are not currently doing, but I would like to start changing this

**Review by JoanFM — CHANGES_REQUESTED** [review]

File: docarray/document/mixins/plot.py [hunk 3]
```diff
@@ -77,6 +77,10 @@ def display(self, from_: Optional[str] = None):
         :param from_: an optional string to decide if a document should display using either the uri or the tensor field.
         """
 
+        if self._is_3d():
+            self.display_3d()
+            return
```
Comment (samsja) [inline docarray/document/mixins/plot.py:82]: do we need this return ?

File: docarray/document/mixins/plot.py [hunk 4]
```diff
@@ -92,7 +96,65 @@ def display(self, from_: Optional[str] = None):
...
 
-    def display_tensor(self):
+    def _is_3d(self) -> bool:
+        """
+        Tells if Document stores a 3D object saved as point cloud or vertices and face.
+        :return: bool.
+        """
+        if self.uri and self.uri.endswith(tuple(MESH_FILE_EXTENSIONS)):
+            return True
+        elif self.tensor is not None and self.tensor.shape[1] == 3:
```
Comment (samsja) [inline docarray/document/mixins/plot.py:106]: Not sure if this work what if I have smth like this

np.zeros(1, 3 ,224 ,224) for an image ? It is still valid image have 3 channels

File: docarray/document/mixins/plot.py (hunk 2 above)
Comment (samsja) [reply docarray/document/mixins/plot.py:124]: I am not a big fan of this check on the name of the class let me search for a better solution

**Review by samsja — CHANGES_REQUESTED** [review]
good job. I added some comments

**Review by samsja — APPROVED** [review]

File: docarray/helper.py [hunk 5]
```diff
@@ -502,3 +502,15 @@ def login(interactive: Optional[bool] = None, force: bool = False, **kwargs):
 def logout():
     """Log out of Hubble account."""
     hubble.logout()
+
+
+def in_notebook() -> bool:
```
Comment (alaeddine-13) [inline docarray/helper.py:507]: There's an is_notebook method in hubble package. We should reuse it

Comment (alaeddine-13) [reply docarray/helper.py:507]: from hubble.utils.notebook import is_notebook

Comment (samsja) [reply docarray/helper.py]: > Do I just adjust this in the setup.py
yes I believe so

**Review by NicholasDunham — APPROVED** [review]

File: docarray/document/mixins/mesh.py [hunk 6]
```diff
@@ -1,15 +1,42 @@
...
+        'ply',
+    ]
+    VERTICES = 'vertices'
+    FACES = 'faces'
 
 
 class MeshDataMixin:
     """Provide helper functions for :class:`Document` to support 3D mesh data and point cloud."""
 
+    def load_mesh(self, force: str = None) -> Union['trimesh.Trimesh', 'trimesh.Scene']:
```
Comment (alaeddine-13) [inline docarray/document/mixins/mesh.py:23]: let's make this private

**Review by alaeddine-13 — APPROVED** [review]
LGTM, just added a minor comment and we're good to go

**Review by samsja — APPROVED** [review]

**Pull request merged**